runtime/nvqir/CircuitSimulator.h

```diff
@@ -134,6 +134,11 @@ class CircuitSimulator {
   /// @brief Allocate `count` qubits.
   virtual std::vector<std::size_t> allocateQubits(const std::size_t count) = 0;
 
+  /// @brief Allocate `count` qubits in a specified input state.
```

Comment:
Requires a new CircuitSimulator overload that accepts the state as input along with the requested number of qubits to allocate.